controllers/argocd/util.go

```diff
@@ -1079,6 +1079,10 @@ func (r *ReconcileArgoCD) setResourceWatches(bldr *builder.Builder, clusterResou
 
 	bldr.Owns(&v1.RoleBinding{})
 
+	nmMapperResourceHandler := handler.EnqueueRequestsFromMapFunc(nmMapper)
+
+	bldr.Watches(&argoproj.NamespaceManagement{}, nmMapperResourceHandler, builder.WithPredicates(namespaceManagementFilterPredicate()))
```

Comment:
With the change I have proposed to `nmMapper` above, I don't think `namespaceManagementFilterPredicate()` is required here.

Reply:
In namespaceManagementFilterPredicate() I am deleting the Rbacs whenever the namespacemanagement field is updated or deleted. Whenever .spec.managedBy field is updated or removed , it will delete the respective rbacs